packages/ember/tests/routing/router_service_test/replaceWith_test.js

```diff
@@ -107,7 +107,7 @@ moduleFor(
         });
     }
 
-    ['@test RouterService#replaceWith with basic query params does not remove query param defaults'](
+    ['@test RouterService#replaceWith with basic query params removes query param defaults'](
```

Comment:
I believe this is incorrect, and while it fixed the issue identified in #19493, this assertion was correct before, per [the RFC](https://rfcs.emberjs.com/id/0095-router-service/#query-parameter-semantics):

> - default values will not be stripped from generated URLs. For example, `urlFor('my-route', { sortBy: 'title' })` will always include `?sortBy=title`, whether or not `title` is the default value of `sortBy`.
> 
> - to explicitly unset a query parameter, you can pass the symbol `Ember.DEFAULT_VALUE` as its value. For example, `transitionTo('my-route', { sortBy: Ember.DEFAULT_VALUE })` will result in a URL that does not contain any `?sortBy=`.

For #19493, if we want to address the issue raised there, I think we need a *separate* test here which captures the case where the user invokes the service method(s) *without* default params, to confirm that we don’t needlessly instantiate and *add* the defaults. @pgengler correctly notes on that issue that the behavior seen there seems contrary to the rest of the intent of the RFC around default query params *but* (a) the relevant specific behavior is not specified by the RFC, (b) the *actual* behavior has been present for a long time, and (c) given that the `transitionTo()` in question is *to the target route*, I think we would end up instantiating the controller *anyway*?

We should have a quick chat about this with Framework folks to nail down what the path forward is. I suspect we need a revert here as a starting point, which I expect will fix #20051, and then to make a decision about the desired path for #19493.

Reply:
Update on the above: see comment [here](https://github.com/emberjs/ember.js/issues/20051#issuecomment-1275061241); this looks unrelated.